backend/server.py

```
@@ -193,31 +212,52 @@ async def login_source(logto_token: str, source_token: str, source: str):
-            "es": "No encontré información relevante sobre ese tema en las fuentes disponibles.",
-            "en": "I couldn't find relevant information about that topic in the available sources.",
-            "gl": "Non atopei información relevante sobre ese tema nas fontes dispoñibles.",
-        }
+        try:
+            result = await app.state.graph.ainvoke(
+                {"messages": [HumanMessage(content=query)]}, config
+            )
+
+        except Exception:
+            logger.exception("Graph invocation failed")
+            raise HTTPException(status_code=500, detail="Internal error processing your request")
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)